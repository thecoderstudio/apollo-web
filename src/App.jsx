import React from 'react';
import { connect } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import styled, { createGlobalStyle, ThemeProvider } from 'styled-components';
import ProtectedRoute from './components/ProtectedRoute';
import NavBar from './components/NavBar';
import Notifications from './components/notifications/Notifications';
import Dashboard from './pages/Dashboard';
import Welcome from './pages/Welcome';
import AgentDetail from './pages/AgentDetail';
import TerminalPage from './pages/TerminalPage';
import Admin from './pages/Admin';
import NotFound from './pages/NotFound';
import { darkTheme } from './theme';

const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=B612:wght@400;700&family=Roboto:wght@300;400&display=swap');

  html, body {
    margin: 0px;
    padding: 0px;
    height: 100%;
    width: 100%;
    font-weight: 500;
    color: ${(props) => props.theme.white};
    background-color: ${(props) => props.theme.black};
  }

  body {
    font-family: 'Roboto', sans-serif;
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: 'B612', sans-serif;
  }
`;

const Content = styled.div`
  height: 100%;
  width: 100%;
`;

function App(props) {
  return (
    <div>
      <ThemeProvider theme={darkTheme}>
        <BrowserRouter>
          <Content>
            {props.authenticated && props.hasChangedPassword && <NavBar />}
            <Switch>
              <ProtectedRoute exact path="/" component={Dashboard} fallbackComponent={Welcome} />
              <ProtectedRoute exact path="/agent/:agentId/shell" component={TerminalPage} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/agent/:agentId" component={AgentDetail} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/admin" component={Admin} fallbackComponent={NotFound} role="admin" />
              <Route component={NotFound} />
            </Switch>
            <Notifications />
            <GlobalStyle />
          </Content>
        </BrowserRouter>
      </ThemeProvider>
    </div>
  );
}

export default connect(
  (state) => ({
    authenticated: state.authenticated,
    hasChangedPassword: state.currentUser.hasChangedInitialPassword
  })
)(App);
