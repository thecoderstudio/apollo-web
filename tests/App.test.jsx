import React from 'react';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { Map } from 'immutable';
import App from '../src/App';

const mockStore = configureStore([]);

function getComponent(store) {
  return renderer.create(
    <Provider store={store}>
      <App />
    </Provider>
  );
}

describe('app', () => {
  it("renders correctly unauthenticated", () => {
    const store = mockStore({
      authenticated: false,
      currentUser: {},
      notifications: new Map({})
    });
    const tree = getComponent(store).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly authenticated", () => {
    const store = mockStore({
      authenticated: true,
      addAgent: {
        selectedArchitecture: "amd64",
        selectedOperatingSystem: "linux"
      },
      agent: [],
      authenticated: true,
      currentUser: {},
      notifications: new Map({})
    });
    const tree = getComponent(store).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
