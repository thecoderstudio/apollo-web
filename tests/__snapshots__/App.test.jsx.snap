// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly authenticated 1`] = `
<div>
  <div
    className="sc-fznJRM ixDFnb"
  >
    <div>
      <div
        className="sc-fznyAO dqfeKx"
      >
        <button
          className="sc-fzpans sc-fzplWN sc-fznKkj dWgsfJ"
          onClick={[Function]}
        >
          Log out
        </button>
      </div>
      <div
        className="sc-fzoLsD sc-fznZeY dctyrU"
      >
        <div
          className="sc-Axmtr cNWFxA"
        >
          <h2
            className="sc-AxmLO jBiGbn"
          >
            Agents
          </h2>
          <ul
            className="sc-fzozJi jqZnKs"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`app renders correctly unauthenticated 1`] = `
<div>
  <div
    className="sc-fznJRM ixDFnb"
  >
    <div
      className="sc-fzqNJr TOXuw"
    >
      <div
        className="sc-fzoyAV jNQZPY"
      >
        <img
          className="sc-fznxsB eqinzM"
          src={[Function]}
        />
        <h1
          className="sc-fzoLag flKWgW"
        >
          Log in to Apollo
        </h1>
        <div
          className="sc-fzqBZW sc-fzoXzr uLdrn"
        >
          <form
            className="sc-fzpjYC cDswVg"
            onSubmit={[Function]}
          >
            <input
              className="sc-fzokOt dbZLPA"
              onChange={[Function]}
              placeholder="Username"
              type="username"
            />
            <input
              className="sc-fzokOt dbZLPA"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
            <button
              className="sc-fzpans bCGSAF"
            >
              Log in
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
